lib/pool/BaseConnectionPool.js

```diff
@@ -102,22 +103,30 @@ class BaseConnectionPool {
    */
   addConnection(opts) {
     if (Array.isArray(opts)) {
-      return opts.forEach((o) => this.addConnection(o));
-    }
+      opts.forEach((o) => this.addConnection(o));
```

Comment:
Any reason why instead of returning this early, you decided to create an if/else block? Wouldn't that make the code harder to read?

Reply:
I see. No reason actually, but let me tell you a story 
At first, i thought that returning a `forEach` was misleading  and removed it. End of story

Actually, i was playing around and it was odd to me that the function returns `void` and `Connection` which is not ideal, imo, but the in typescript definition it states that `addConnection` always returns `Connection`


I tried using Array.`map` and return array of connections,
```ts
addConnection(...): Connection | Connection[]
```
 but the tests failed and they were many, so I reverted everything back and the else block somehow stuck around :D

I'll remove the else block
But i'm moving the return statement to next line :P 
```ts
if (Array.isArray(opts)) {
      opts.forEach((o) => this.addConnection(o));
      return;
 }
```